django/db/models/lookups.py

```diff
@@ -489,7 +433,19 @@ def get_bound(self):
         )
 
 
-@YearTransform.register_lookup
+class YearExact(YearLookup):
+    lookup_name = 'exact'
+
+    def as_sql(self, compiler, connection):
+        # We will need to skip the extract part and instead go
+        # directly with the originating field, that is self.lhs.lhs.
+        lhs_sql, params = self.process_lhs(compiler, connection, self.lhs.lhs)
+        rhs_sql, rhs_params = self.process_rhs(compiler, connection)
+        bounds = self.year_lookup_bounds(connection, rhs_params[0])
+        params.extend(bounds)
+        return '%s BETWEEN %%s AND %%s' % lhs_sql, params
+
+
```

Comment:
They are registered against YearLookup (https://github.com/django/django/pull/6243/files#diff-d4a6ef1ea81ea8f2b3d3c53d39463b90R92) but I'll make sure there are tests that verify the behaviour. Thanks for pointing that out.